server/src/deployment/gcp/gcpdisk.ts

```diff
@@ -10,37 +10,90 @@
 import {Disk, DiskManager} from "../disks";
 import Compute from '@google-cloud/compute';
 
-import {arcsKeyFor} from "../utils";
+import {ARCS_NODE_LABEL, arcsKeyFor, waitForGcp} from "../utils";
 
 
+import common from "@google-cloud/common";
+import {GCE_PERSISTENT_DISK_TYPE, GCP_ZONE} from "./gcp-constants";
+
 /**
  * Represents disk storage provisioned on a cloud provider.
  */
 class GCPDisk implements Disk {
 
   diskApi: Compute.Disk;
+  private computeApi: Compute;
 
-  constructor(diskApi) {
+  constructor(compute: Compute, diskApi) {
+    this.computeApi = compute;
     this.diskApi = diskApi;
   }
 
-  isAttached(): boolean {
+  async isAttached(): Promise<boolean> {
+    const [metadata] = await this.diskApi.getMetadata();
+    if (metadata['users'] && metadata['users'].length) {
```

Comment:
nit: Can just `return metadata['users'] && metadata['users'].length`

I think you could also do:

```
const {users} = metadata;
return users && users.length;
```